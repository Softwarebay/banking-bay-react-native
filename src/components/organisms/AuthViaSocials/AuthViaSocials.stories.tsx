import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { AuthViaSocials } from './AuthViaSocials';

storiesOf('Organisms/AuthViaSocials', module)
	.addDecorator((getStory) => <ViewStory>{getStory()}</ViewStory>)
	.add('Default', () => (
		<AuthViaSocials
			onPressFacebook={action('onPressFacebook')}
			onPressGooglePlus={action('onPressGooglePlus')}
			onPressTwitter={action('onPressTwitter')}
		/>
	));
