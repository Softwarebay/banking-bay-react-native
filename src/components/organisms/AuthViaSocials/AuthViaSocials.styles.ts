import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	container: {
		alignItems: 'center',
		flexDirection: 'row',
		justifyContent: 'center',
	},
	iconBtn: {
		height: 50,
		marginLeft: 10,
		width: 50,
	},
});
