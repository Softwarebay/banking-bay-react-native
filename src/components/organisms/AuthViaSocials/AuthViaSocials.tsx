import { Flex, IconButton } from 'components/atoms';
import { colors } from 'theme';

import { styles } from './AuthViaSocials.styles';

interface AuthViaSocialItem {
	name: string;
	btnColor: string;
	iconName: string;
	iconColor: string;
	onPress: () => void;
}

interface Props {
	onPressFacebook: () => void;
	onPressTwitter: () => void;
	onPressGooglePlus: () => void;
}

export const AuthViaSocials = ({
	onPressFacebook,
	onPressGooglePlus,
	onPressTwitter,
}: Props) => {
	const authViaSocials: AuthViaSocialItem[] = [
		{
			name: 'Facebook',
			iconName: 'facebook',
			iconColor: colors.white,
			btnColor: '#3B5999',
			onPress: onPressFacebook,
		},
		{
			name: 'Twitter',
			iconName: 'twitter',
			iconColor: colors.white,
			btnColor: '#55ACEE',
			onPress: onPressTwitter,
		},
		{
			name: 'GooglePlus',
			iconName: 'google-plus',
			iconColor: colors.white,
			btnColor: '#DD4B39',
			onPress: onPressGooglePlus,
		},
	];

	return (
		<Flex direction="row" style={styles.container}>
			{authViaSocials.map(
				({ name, iconName, iconColor, btnColor, onPress }) => (
					<IconButton
						key={name}
						btnStyle={[styles.iconBtn, { backgroundColor: btnColor }]}
						iconColor={iconColor}
						iconName={iconName}
						testID={`${name}-btn-test-id`}
						onPress={onPress}
					/>
				)
			)}
		</Flex>
	);
};
