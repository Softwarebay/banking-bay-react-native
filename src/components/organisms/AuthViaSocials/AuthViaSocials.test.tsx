import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { AuthViaSocials } from './AuthViaSocials';

import type { Props } from './AuthViaSocials';

describe('AuthViaSocials', () => {
	const mockOnPressFacebook = jest.fn<Props['onPressFacebook'], []>();
	const mockOnPressGooglePlus = jest.fn<Props['onPressGooglePlus'], []>();
	const mockOnPressTwitter = jest.fn<Props['onPressTwitter'], []>();

	const props: Props = {
		onPressFacebook: mockOnPressFacebook,
		onPressGooglePlus: mockOnPressGooglePlus,
		onPressTwitter: mockOnPressTwitter,
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<AuthViaSocials {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render 3 icon buttons', () => {
		render(<AuthViaSocials {...props} />);

		const authButtons = screen.getAllByTestId(
			/(facebook|twitter|googlePlus)-icon-btn-test-id/gi
		);

		expect(authButtons).toHaveLength(3);
	});

	it('should handle press events on Facebook icon', () => {
		render(<AuthViaSocials {...props} />);

		const facebookIconBtn = screen.getByTestId(/facebook-icon-btn-test-id/i);

		fireEvent.press(facebookIconBtn);

		expect(mockOnPressFacebook).toHaveBeenCalledTimes(1);
		expect(mockOnPressGooglePlus).not.toHaveBeenCalledTimes(1);
		expect(mockOnPressTwitter).not.toHaveBeenCalledTimes(1);
	});

	it('should handle press events on Google+ icon', () => {
		render(<AuthViaSocials {...props} />);

		const googlePlusIconBtn = screen.getByTestId(
			/googlePlus-icon-btn-test-id/i
		);

		fireEvent.press(googlePlusIconBtn);

		expect(mockOnPressGooglePlus).toHaveBeenCalledTimes(1);
		expect(mockOnPressFacebook).not.toHaveBeenCalledTimes(1);
		expect(mockOnPressTwitter).not.toHaveBeenCalledTimes(1);
	});

	it('should handle press events on Twitter icon', () => {
		render(<AuthViaSocials {...props} />);

		const twitterIconBtn = screen.getByTestId(/twitter-icon-btn-test-id/i);

		fireEvent.press(twitterIconBtn);

		expect(mockOnPressTwitter).toHaveBeenCalledTimes(1);
		expect(mockOnPressFacebook).not.toHaveBeenCalledTimes(1);
		expect(mockOnPressGooglePlus).not.toHaveBeenCalledTimes(1);
	});
});
