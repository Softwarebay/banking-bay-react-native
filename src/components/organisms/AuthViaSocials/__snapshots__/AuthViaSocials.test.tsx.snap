// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthViaSocials should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={20}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Array [
          Object {
            "borderRadius": 100,
            "height": 60,
            "marginLeft": 10,
            "width": 60,
          },
          Object {
            "backgroundColor": "#3B5999",
          },
        ],
        false,
      ]
    }
    testID="Facebook-icon-btn-test-id"
  >
    <SvgMock
      color="#FFFFFF"
      height={27}
      testID="facebook-icon-test-id"
      width={27}
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={20}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Array [
          Object {
            "borderRadius": 100,
            "height": 60,
            "marginLeft": 10,
            "width": 60,
          },
          Object {
            "backgroundColor": "#55ACEE",
          },
        ],
        false,
      ]
    }
    testID="Twitter-icon-btn-test-id"
  >
    <SvgMock
      color="#FFFFFF"
      height={27}
      testID="twitter-icon-test-id"
      width={27}
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={20}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Array [
          Object {
            "borderRadius": 100,
            "height": 60,
            "marginLeft": 10,
            "width": 60,
          },
          Object {
            "backgroundColor": "#DD4B39",
          },
        ],
        false,
      ]
    }
    testID="GooglePlus-icon-btn-test-id"
  >
    <SvgMock
      color="#FFFFFF"
      height={27}
      testID="google-plus-icon-test-id"
      width={27}
    />
  </View>
</View>
`;
